crd/src/lib.rs

```diff
@@ -70,3 +96,30 @@ pub struct SelectorAndConfig<T> {
 #[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct KafkaConfig {}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use indoc::indoc;
+    use rstest::rstest;
+
+    #[rstest]
+    #[case::no_scala_version(
+    indoc! {"
+          kafka_version: 2.6.0
+      "},
+    "2.13-2.6.0"
+    )]
+    #[case::with_scala_version(
+    indoc! {"
+          kafka_version: 2.8.0
+          scala_version: 2.12
+      "},
+    "2.12-2.8.0"
+    )]
```

Comment:
`indoc!` is not necessary here.

```suggestion
    #[case::with_scala_version(
        "
          kafka_version: 2.8.0
          scala_version: 2.12
        ",
        "2.12-2.8.0"
    )]
```